### PR TITLE
Add parimutuel docs

### PR DESCRIPTION
**Pull request by maltekliemann**

### CODE REVIEW

**Review by Chralt98 — COMMENTED** [review]
Good thanks! Please make sure to explain that Parimutuel markets are supposed to be used in situations, where the odds don't change. Example: close the parimutuel market before the horse race even starts.

File: docs/learn/parimutuel.md [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+used on Zeitgeist for extra casual markets.
+
+These are "losers pay winners" market makers: Any informant can bet any amount
+at any time. Their bet amount goes into the _pot_ and they receive tokens which
+represent their share of the pot. After the market is resolved, the entire pot
+is distributed amongst those who wagered on the outcome that materialized,
+proportional to what their share of the pot is.
+
+Although parimutuels work with scalar markets, Zeitgeist currently only supports
+parimutuel for categorical markets.
```
[the remark this answers is not in the document]
Comment (maltekliemann) [reply docs/learn/parimutuel.md:18]: I don't think we need to explain why we don't implement certain features. I also disagree with some of what you write. I don't think there's any problem with scalar markets in conjunction with parimutuels that you don't also have with categorical markets.

File: docs/learn/parimutuel.md [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+Unlike automatic market makers (AMM) or continuous double-auction (CDA), the
+parimutuel market maker does not require any liquidity, and shares the property
+of AMM that it can fill any order at any time. It is essentially a "bring your
+own liquidity" market maker.
+
+However, it does suffers several disadvantages compared to the other mechanisms
+on Zeitgeist:
+
+- The odds are not fixed when tokens are bought. For example, if an informants
```
[the remark this answers is not in the document]
Comment (maltekliemann) [reply docs/learn/parimutuel.md]: "The odds are not fixed when tokens are bought" refers to the payout odds.

File: docs/learn/parimutuel.md [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+  A particularly vexing symptom of this problem is that, if a market becomes
+  trivialized (some outcome $X$ has materialized before the end of the market)
+  and at least two agents have bet on the winning outcome, then it's a winning
+  strategy to keep pumping more money into the market to dilute the other
+  agent's stake.
+
+- No selling of contracts. Once you've bought a contract, you have to hold it.
+  You can't just take back your bet. This means that parimutuels are really only
+  suited for markets which resolve very quickly.
```
[the remark this answers is not in the document]
Comment (maltekliemann) [reply docs/learn/parimutuel.md:77]: Since we don't discuss dynamic parimutuels, I don't think there's any point in mentioning them here.

File: docs/learn/parimutuel.md [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+
+As such, parimutuel markets are perfectly suited for short-lived markets where
+the market's outcome is published at a predefined time.
```
[the remark this answers is not in the document]
Comment (maltekliemann) [reply docs/learn/parimutuel.md:80]: "[...] the probabilities of the outcomes don't change during the active market period." And how would we know that? Seems far-fetched.

File: docs/learn/parimutuel.md [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+
+## Parimutuel Markets on Zeitgeist
+
+### Betting
+
+Every parimutuel market uses an special account as pot. If an informant places a
+bet, they send `x` units of collateral to the pot and receive `x` units of the
+corresponding type of _parimutuel shares_. Informants must observe a minimum bet
+size defined in the parimutuel pallet when placing their bets.
```
[the remark this answers is not in the document]
Comment (maltekliemann) [reply docs/learn/parimutuel.md]: What else would the minimum bet size be denominated in if the bets are made in collateral?

File: docs/learn/parimutuel.md [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+
+<!-- TODO External fees to be defined in the general section on Zeitgeist markets in a later PR. -->
+
+External fees are paid when users buy parimutuel shares in the usual fashion: If
+Alice buys parimutuel shares for a certain amount, then the external fees are
+deducted from this amount before the rest of the transaction is executed. The
+amount Alice is left with after fees are deducted must satisfy the minimum bet
+size requirement.
```
[the remark this answers is not in the document]
Comment (maltekliemann) [reply docs/learn/parimutuel.md]: 55b7262cf34d7afc75d08eba9e44be138a06e603

File: docs/learn/parimutuel.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (maltekliemann) [reply docs/learn/parimutuel.md]: Thought about this some more and reconsidered my opinion. Downgraded this to informants believing that the odds don't change.

79de7187a73db5b70e3c4a81c5b9066f5c950011

File: docs/learn/parimutuel.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (Chralt98) [reply docs/learn/parimutuel.md:18]: > I don't think we need to explain why we don't implement certain features.

Ok.

> I don't think there's any problem with scalar markets in conjunction with parimutuels that you don't also have with categorical markets.

Please think the reward process through with scalar markets. I thought we came to the same conclusion in our process and for that reason we decided to remove scalar markets for Parimutuels. Didn't we?

For categorical markets, if you buy the winning outcome, you can not lose any base asset. For scalar markets you can lose base asset, because there is no clear winner of LONG or SHORT, because the pot is divided by the actual scalar value between range start and range end.

**Review by Chralt98 — APPROVED** [review]

File: docs/learn/parimutuel.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (maltekliemann) [reply docs/learn/parimutuel.md:18]: There's a clear winning strategy for scalar markets if your prediction is correct: If you think that the long correctness is c, and thus the short correctness is (1-c), then buy xc units of LONG and x(1-c) units of SHORT. The payoff is x(cr_l + (1 - c)r_s). You can know show that this greater or equal to x and equality holds if and only if everybody else is also buying at the same ratio.

The approach to do this is to define y = a_s/a_l and then calculate

cr_l + (1-c)r_s = c^2 * (1 + y) + (1 - c)^2 * (1 + 1/y).

You can now define f(c, y) = c^2 * (1 + y) + (1 - c)^2 * (1 + 1/y) and take first and second partial derivatives to see that in the strip (0, 1) \times (0, \infty), we have local minima at c = 1/(y + 1), and f(1/(y + 1), y) = 1, which proves the assertion (takes a bit of calculation to get through this).

TL;DR: Betting on parimutuel scalar markets just works a bit different than with AMM or CDA. You don't bet based on market state because... well, there is no market state.

[the remark this answers is not in the document]
Comment (Chralt98) [reply docs/learn/parimutuel.md:18]: The doctor himself spoke.
